### PR TITLE
Add tests for New-Object constrained mode

### PR DESCRIPTION
**Pull request by chunqingchen**
addresses #4154

After reviewing the code coverage file and existed tests, creating object under constrained mode is not covered.

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/New-Object.Tests.ps1 [hunk 1]
```diff
@@ -1,3 +1,101 @@
...
+
+            if (revertLockdownMode)
+            {
+                Environment.SetEnvironmentVariable("__PSLockdownPolicy", null, EnvironmentVariableTarget.Machine);
+            }            
+        }
+    }
+'@
+
+    if (-not (Get-Command Invoke-LanguageModeTestingSupportCmdlet -ea Ignore))
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/New-Object.Tests.ps1]: Use `-ErrorAction` instead

File: test/powershell/Modules/Microsoft.PowerShell.Utility/New-Object.Tests.ps1 [hunk 2]
```diff
@@ -164,6 +262,38 @@ Describe "New-Object DRT basic functionality" -Tags "CI" {
 		$result = New-Object -TypeName PSObject -property @{foo=123}
 		$result.foo | Should Be 123
 	}
+
+    It "New-Object object cannot create COM object under constrained mode" -skip:(-not $IsWindows) {
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/New-Object.Tests.ps1]: I think we should add these tests here: https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1#L226

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by chunqingchen** [comment]
@adityapatwardhan your comment is resolved

File: test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1 [hunk 3]
```diff
@@ -905,6 +905,41 @@ try
         }
     }
 
+    Describe "New-Object constrained language test" -Tags "CI" {
+
+        It "New-Object object cannot create COM object under constrained mode" -skip:(-not $IsWindows) {
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1]: Can this be another test case in the same context  https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1#L277

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by adityapatwardhan** [comment]
@chunqingchen  Please resolve conflict in 'test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1 '

**Comment by adityapatwardhan** [comment]
@chunqingchen ping..

**Comment by chunqingchen** [comment]
I think this is duplicated in the constrained tests. I think we can just close this pr without submitting it.

**Comment by adityapatwardhan** [comment]
@chunqingchen  I think you can close the PR. Maybe put a link to the code where we have coverage.

**Pull request closed without merging**